packages/core/src/PartialExecutionUtils/__tests__/findSubgraph.test.ts

```diff
@@ -0,0 +1,203 @@
+
+		expect(subgraph).toEqual(graph);
+	});
+
+	//                     ►►
+	//  ┌───────┐         ┌───────────┐
+	//  │       ├────────►│           │      ┌────┐
+	//  │trigger│         │destination├─────►│node│
+	//  │       ├────────►│           │      └────┘
+	//  └───────┘         └───────────┘
+	test('disregard nodes after destination', () => {
+		const trigger = createNodeData({ name: 'trigger' });
+		const destination = createNodeData({ name: 'destination' });
+		const node = createNodeData({ name: 'node' });
+
+		const graph = new DirectedGraph()
+			.addNodes(trigger, destination, node)
+			.addConnections({ from: trigger, to: destination }, { from: destination, to: node });
+
+		const subgraph = findSubgraph(graph, destination, trigger);
+
+		expect(subgraph).toEqual(
+			new DirectedGraph()
+				.addNodes(trigger, destination)
+				.addConnections({ from: trigger, to: destination }),
+		);
+	});
+
+	//                     XX
+	//  ┌───────┐         ┌────────┐       ►►
+	//  │       ├────────►│        │      ┌───────────┐
+	//  │trigger│         │disabled├─────►│destination│
+	//  │       ├────────►│        │      └───────────┘
+	//  └───────┘         └────────┘
+	test('skip disabled nodes', () => {
+		const trigger = createNodeData({ name: 'trigger' });
+		const disabled = createNodeData({ name: 'disabled', disabled: true });
+		const destination = createNodeData({ name: 'destination' });
+
+		const graph = new DirectedGraph()
+			.addNodes(trigger, disabled, destination)
+			.addConnections({ from: trigger, to: disabled }, { from: disabled, to: destination });
+
+		const subgraph = findSubgraph(graph, destination, trigger);
+
+		expect(subgraph).toEqual(
+			new DirectedGraph()
+				.addNodes(trigger, destination)
+				.addConnections({ from: trigger, to: destination }),
+		);
+	});
+
+	//                                ►►
+	//  ┌───────┐       ┌─────┐      ┌─────┐
+	//  │Trigger├───┬──►│Node1├───┬─►│Node2│
+	//  └───────┘   │   └─────┘   │  └─────┘
+	//              │             │
+	//              └─────────────┘
+	test('terminates when called with graph that contains cycles', () => {
+		//
+		// ARRANGE
+		//
+		const trigger = createNodeData({ name: 'trigger' });
+		const node1 = createNodeData({ name: 'node1' });
+		const node2 = createNodeData({ name: 'node2' });
+		const graph = new DirectedGraph()
+			.addNodes(trigger, node1, node2)
+			.addConnections(
+				{ from: trigger, to: node1 },
+				{ from: node1, to: node1 },
+				{ from: node1, to: node2 },
+			);
+
+		//
+		// ACT
+		//
+		const subgraph = findSubgraph(graph, node2, trigger);
+
+		//
+		// ASSERT
+		//
+		expect(subgraph).toEqual(graph);
+	});
+
+	//                ►►
+	//  ┌───────┐     ┌─────┐
+	//  │Trigger├──┬─►│Node1│
+	//  └───────┘  │  └─────┘
+	//             │
+	//  ┌─────┐    │
+	//  │Node2├────┘
+	//  └─────┘
+	test('terminates when called with graph that contains cycles', () => {
+		//
+		// ARRANGE
+		//
```

Comment:
nit: Could we stick with single line comments `// ARRANGE`, `// ACT` and `// ASSERT`?